File: Bit Manipulation/3307.Find the K-th Character in String Game II/Solution.java
class Solution {
    public char kthCharacter(long k, int[] operations) {
        int m = 0;
        while (1L << m < k) {
            m++;
        }
        long pos = k, cnt = 0;
        while(m > 0) {
            long half = 1L << (m - 1);
            if (pos > half) {
                cnt += operations[m - 1];
                pos -= half;
            }
            m--;
        }
        return (char)(cnt % 26 + 'a');
    }
}

File: Bit Manipulation/3307.Find the K-th Character in String Game II/Solution.py
class Solution:
    def kthCharacter(self, k: int, operations: List[int]) -> str:
        m = 0
        while 1 << m < k:
            m += 1
        pos = k
        cnt = 0
        while m > 0:
            half = 1 << (m - 1)
            if pos > half:
                pos -= half
                if operations[m - 1]:
                    cnt += 1
            m -= 1
        return chr(cnt % 26 + ord('a'))